SCClassLibrary/Common/Control/Server.sc

```diff
@@ -348,23 +370,42 @@ Server {
 	}
 
 	initTree {
-		nodeAllocator = NodeIDAllocator(clientID, options.initialNodeID);
-		this.sendMsg("/g_new", 1, 0, 0);
+		this.newNodeAllocators;
+		this.sendMsg("/g_new", this.defaultGroup.nodeID, 0, 0);
 		tree.value(this);
 		ServerTree.run(this);
 	}
 
 	/* id allocators */
 
+	// private, called from server notify response with next free clientID
 	clientID_ { |val|
+		var failstr = "Server % couldn't set client id to: % - %. clientID is still %.";
+
+		if(val == clientID) {
+			// no need to change
+			^this
+		};
 		if(val.isInteger.not) {
-			"Server % couldn't set client id to: %".format(name, val.asCompileString).warn;
+			failstr.format(name, val.cs, "not an Integer", clientID).warn;
 			^this
 		};
-		if(clientID != val) {
-			clientID = val;
-			this.newAllocators;
-		}
+		if (val < 0) {
+			failstr.format(name, val.cs, "less than minimum 0", clientID).warn;
+			^this
+		};
+		if (val >= this.numClients) {
+			failstr.format(name,
+				val.cs,
+				"greater than server.numClients % minus 1 allows".format(this.numClients),
```

Comment:
this is a bit obscure — what would you think of combining these two conditions and saying "outside of allowed clientID range (0-%)"?

Reply:
yes, clearer, done.